### ISSUE TITLE
Ast_pattern: example in doc doesn't type

### ISSUE THREAD

**Issue by pveber** [ocaml-ppx/ppxlib#178]
[An example in the doc](https://github.com/ocaml-ppx/ppxlib/blob/a3878ab32e493e03286bc9a8d72d27c8c2f3d178/src/ast_pattern.mli#L57) doesn't type:
```ocaml
let payload_pattern name expr =
        let open Ast_pattern in
        pstr (pstr_value nonrecursive (value_binding ~pat:(pstring __) ~expr:__) ^:: nil)
```
Here's the compiler's complaint:
```
This expression has type (value_binding, label -> expression -> 'a, 'a) t
but an expression was expected of type (value_binding list, 'b, 'c) t
Type value_binding is not compatible with type value_binding list
```
Here's what I think is a possible fix:
```ocaml
let payload_pattern name expr =
  let open Ast_pattern in
  pstr (pstr_value nonrecursive ((value_binding ~pat:(pstring __) ~expr:__) ^:: nil) ^:: nil)
```

**Comment by Kakadu** [comment]
@pitag-ha Let's close this?

**Comment by panglesd** [comment]
Closed by #365.
